source/fundamentals/crud/compound-operations.txt

```diff
@@ -44,43 +44,50 @@ These methods accept an optional ``options`` object with
+of these methods. To learn more about this option, see the
+:ref:`includeResultMetadata Option <node-compound-metadata-option>`
+section of this guide.
 
-   Starting in version 5.7, you can set the ``includeResultMetadata``
-   setting in the ``options`` object to specify the return type for each
-   of these methods.
-   
-   Starting in version 6.0, this setting defaults to ``false``, which
-   means that each method returns the matched document. If no
-   document is matched, each method returns ``null``. If you set
-   ``includeResultMetadata`` to ``true``, the method returns a
-   ``ModifyResult`` type that contains the found document and additional
-   metadata.
+The ``findOneAndUpdate()`` and ``findOneAndDelete()`` methods take the
+``returnDocument`` setting, which specifies if the method returns the
+pre-update or post-update version of the modified document.
+
+.. _node-compound-metadata-option:
 
-   Suppose a collection contains only the following document:
+includeResultMetadata Option
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The ``includeResultMetadata`` option determines the return type of the
+compound methods.
+   
+This setting defaults to ``false``, which
+means that each method returns the matched document. If no
+document is matched, each method returns ``null``. If you set
+``includeResultMetadata`` to ``true``, the method returns a
+``ModifyResult`` type that contains the found document and additional
+metadata.
 
-   .. code-block:: json
+Suppose a collection contains only the following document:
 
-      { _id: 1, x: "on" }
+.. code-block:: json
 
-   The following code shows how the value of the
-   ``includeResultMetadata`` option changes the return type of
-   the ``findOneAndDelete()`` method:
+   { _id: 1, x: "on" }
 
-   .. code-block:: js
+The following code shows how the value of the
+``includeResultMetadata`` option changes the return type of
+the ``findOneAndDelete()`` method:
 
-      // default behavior
-      // returns { _id: 1, x: 'on' }
-      await coll.findOneAndDelete({ x: "on" });
+.. code-block:: js
 
-      // returns { lastErrorObject: { n: 1 }, value: { _id: 1, x: 'on' }, ok: 1, ... }
-      await coll.findOneAndDelete({ x: "on" }, { includeResultMetadata: true });
+   // default behavior
+   // returns { _id: 1, x: 'on' }
+   await coll.findOneAndDelete({ x: "on" });
 
-      // no document matched
-      // returns null
-      await coll.findOneAndDelete({ x: "off" });
+   // returns { lastErrorObject: { n: 1 }, value: { _id: 1, x: 'on' }, ok: 1, ... }
+   await coll.findOneAndDelete({ x: "on" }, { includeResultMetadata: true });
 
-You can set the ``returnDocument`` setting in the ``options`` object for the
-``findOneAndUpdate()`` and ``findOneAndDelete()`` methods, which lets
-you specify if the method returns the pre-update or post-update version
-of the modified document.
+   // no document matched
+   // returns null
+   await coll.findOneAndDelete({ x: "off" });
```

Comment:
S: this might be easier to read as a table, with a column for the description and a column for code examples. You could also format the code examples as IO code blocks so the returned data is in the output section rather than a comment. Up to you though

Reply:
good idea on including a table! Will implement.